### ISSUE TITLE
New cop that check if find_by(*) works fine

### ISSUE THREAD

**Issue by konumashuri** [bbatsov/rubocop#3994]
I'd like to suggest a new cop, which detects misusing of `find_by` method (ie: `find_by(some_id)`) and points out it should be `find_by(id: some_id)`
I think it's really hard to find that misusing.

Note: I'm not sure if this is rubocop's business, so if it's not then tell me so please. Thanks :)

**Comment by Drenmi** [comment]
The idea is good, but it is very hard to do this without false positives in RuboCop, because the variable passed to `#find_by` could reference a hash, which is perfectly fine. 😕

**Comment by konumashuri** [comment]
Thank you for your replying.
I see... Can I try to add the cop and submit pull request? I'm thinking this cop will check if  argument is `id: some_id` or `{ id: some_id }`.

**Comment by bbatsov** [comment]
Sure.

**Comment by koic** [comment]
In Rails 5.0 and higher, giving an `Integer` object as an argument to `find_by` causes an error.

```ruby
> Model.find_by(1)
ArgumentError: Unsupported argument type: 1 (Integer)
```

https://github.com/rails/rails/commit/b27fb2d8e76

Currently Rails 4.2 is released only for severe security maintenance, but it will soon be an EOL.

http://guides.rubyonrails.org/maintenance_policy.html
http://weblog.rubyonrails.org/2017/9/27/Rails-4-2-10-released

I thought about taking over this implementation of cop. However, it may be less useful in the near future and I canceled it. Perhaps it may be good to close this issue.

**Comment by koic** [comment]
As mentioned above comment, this is not a problem if Rails is upgraded to 5.0 or higher. and I think that is the main solution.

And I think that minimum supported Rails version by Rails department will go up in the future. So this cop will be short in its life cycle. I'd like to close this issue If there is no objection.
